resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala

```diff
@@ -140,13 +140,6 @@ private[spark] class Client(
           throw e
       }
 
-      if (waitForAppCompletion) {
```

Comment:
Why is this not needed anymore? If we enable cluster mode we still want the same behavior defined here right?

Reply:
mmh didn't change that behavior. Let's wait the next push and see the diff. master is evolving and numerous refactoring does not make this PR easy to merge with.

Reply:
Right, I have added back that section... Thx @tnachen